contracts/contracts/UmbraRouter.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraRouter {
+  address internal constant umbra = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
+
+  struct SendEth {
+    address payable receiver;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+
+  struct SendToken {
+    address receiver;
+    address tokenAddr;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+
+  function batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) public payable {
+    uint256 balance = msg.value;
+
+    for (uint256 i = 0; i < _params.length; i++) {
+      //amount to be sent per receiver
+      uint256 _amount = _params[i].amount;
+      require(balance >= _amount, "not enough msg.value");
+      balance -= _amount;
+
+      bytes memory data =
+        abi.encodeWithSelector(
+          IUmbra.sendEth.selector,
+          _params[i].receiver,
+          _tollCommitment,
+          _params[i].pkx,
+          _params[i].ciphertext
+        );
+
+      (bool success, ) = umbra.call{value: _amount}(data);
+      require(success, "sendEth call failed");
```

Comment:
Instead of doing this manually, we can make things cleaner by leveraging our interface. This is the reason for making the `umbra` variable of type `IUmbra` instead of `address`

```suggestion
      umbra.sendEth(_params[i].receiver, _tollCommitment, _params[i].pkx, _params[i].ciphertext);
```

Reply:
I thought you suggested using abi.encodeWithSelector. https://github.com/ScopeLift/umbra-protocol/pull/311#discussion_r837951449 What would be a situation where you might want to use selectors?

Reply:
Yea sorry that was unclear—a few lines down when I saw what's going on I added a separate about using `umbra.sendETH` instead. In context of that specific line, `encodeWithSelector` was a better choice, but in context of the contract, `umbra.sendEth` is better.

Here's an example where a [timelock](https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/Timelock.sol#L95) contract uses `abi.encodePacked` because it doesn't know the sig/calldata until runtime. They could have also done `abi.encodeWithSignature`

Reply:
(feel free to mark this one resolved whenever you see fit)